### PR TITLE
Bug fix 3077

### PR DESCRIPTION
**Pull request by sandy081**
Provide following support for css variables
- Coloring
- Intellisense
- Peek and Go to

Also provide the same for css variables in sass and less

Unit test covered

### CODE REVIEW

File: build/gulpfile.common.js [hunk 1]
```diff
@@ -40,10 +40,10 @@ exports.loaderConfig = function (emptyPaths) {
 			paths: {
 				'vs/extensions': 'extensions'
 			}
-		}
+		},
+		nodeModules: emptyPaths||[]
```
Comment (aeschli) [inline build/gulpfile.common.js]: Is this change related to 3077?

File: build/gulpfile.vscode.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ var baseModules = [
 	'applicationinsights', 'assert', 'child_process', 'chokidar', 'crypto', 'emmet',
 	'events', 'fs', 'getmac', 'glob', 'graceful-fs', 'http', 'http-proxy-agent',
 	'https', 'https-proxy-agent', 'iconv-lite', 'electron', 'net',
-	'os', 'path', 'readline', 'sax', 'semver', 'stream', 'string_decoder', 'url',
+	'os', 'path', 'pty.js', 'readline', 'sax', 'semver', 'stream', 'string_decoder', 'url', 'term.js',
```
Comment (aeschli) [inline build/gulpfile.vscode.js]: Is this change related to 3077?

File: extensions/node-debug/node-debug.azure.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"account": "monacobuild",
 	"container": "debuggers",
-	"zip": "624d700/node-debug.zip",
+	"zip": "3c7ed19/node-debug.zip",
```
Comment (aeschli) [inline extensions/node-debug/node-debug.azure.json]: Is this change related to 3077?

File: build/gulpfile.vscode.js (hunk 2 above)
Comment (Tyriar) [inline build/gulpfile.vscode.js]: This is related to integrated terminal just being disabled, pty.js and term.js should not be in `baseModules` on master atm.

Comment (aeschli) [inline build/gulpfile.vscode.js]: Looks like Sandeeps branch got messed up. Thanks Daniel for commenting.

**Comment by sandy081** [comment]
@aeschli 

Yeah this branch has got messed up.. Looks like I did a sync and it merged all from master..
I created a new branch sandy/css-variables and made a new pull request..
I will delete this branch and please do review in the new one..

Sorry for inconvenience
